src/features/unified-share-modal/UnifiedShareForm.js

```diff
@@ -125,6 +196,22 @@ class UnifiedShareForm extends React.Component<USFProps, State> {
             numOfInviteeGroups: groupIDs.length,
         };
 
+        const hasJustificationReason = !!justificationReason;
+        const hasRestrictedExternalInvitees = !isEmpty(restrictedExternalEmails);
+        const shouldSubmitJustificationReason =
+            hasJustificationReason && hasRestrictedExternalInvitees && isCollabRestrictionJustificationAllowed;
+
+        if (shouldSubmitJustificationReason) {
+            params = {
+                ...params,
+                classificationLabelId,
```

Comment:
Noting that these will not be passed directly to the send invites api since the justification will be submitted as a separate request. I've been considering whether or not it'd be better to pass this as a second params object, but thought it might be cleaner this way since `ContentSharing` is already converting the params into the format it needs, for example.

Also note that `ContentSharing` will not reach this code path since it's not wired to trigger the justification flow. This is expected since this flow will not be initially supported for platform.